src/main/java/com/dev/cinema/controller/OrderController.java

```diff
@@ -34,15 +36,19 @@ public OrderController(OrderService orderService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto complete(@RequestParam Long id) {
-        return orderMapper
-                .toDto(orderService.completeOrder(shoppingCartService
-                        .getByUser(userService.get(id))));
+    public OrderResponseDto complete(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
```

Comment:
if you wanna use that, then you can write like: 
```suggestion
        UserDetails userDetails = authentication.getName()
```

Reply:
I don;t know if that is right option. `getName()` returns String. And `getPrincipal()` returns an Object of `UserDetails`. Or you mean get email with `getName()` method without creating `userDetails`?

Reply:
yes, we can get email with getName() method without creating userDetails